core/src/main/java/org/apache/calcite/sql/validate/SqlUserDefinedTableMacro.java

```diff
@@ -147,6 +147,8 @@ private static Object getValue(SqlNode right) throws NonLiteralException {
         builder2.put(getValue(key), getValue(value));
       }
       return builder2.build();
+    case CAST:
+      return getValue(((SqlCall) right).operand(0));
```

Comment:
> "All arguments of call to macro s.str should be literal"

From above exception message from JIRA description, I understands that only literal is allowed for the macro.
If we support CAST, do we target to support some other expressions ? like `select * from table("s"."str"(MAP['a', 1, 'baz', 2], 0 + 1)) as t(n)` ? Why we take `CAST` into special consideration ?

Reply:
Because, `CAST` may happen implicitly, see `testUserDefinedTableFunction4` in `TableFunctionTest`.
We can also support  other type of expressions, like `0 + 1`, or even function call `max(1, 2)`, but I think that's another topic.